src/driver/database-connection.ts

```diff
@@ -1,15 +1,21 @@
 import { CompiledQuery } from '../query-compiler/compiled-query.js'
+import { ExecuteQueryOptions } from '../query-executor/query-executor.js'
 
 /**
  * A single connection to the database engine.
  *
  * These are created by an instance of {@link Driver}.
  */
 export interface DatabaseConnection {
-  executeQuery<R>(compiledQuery: CompiledQuery): Promise<QueryResult<R>>
+  executeQuery<R>(
+    compiledQuery: CompiledQuery,
+    options?: ExecuteQueryOptions,
+  ): Promise<QueryResult<R>>
+
   streamQuery<R>(
     compiledQuery: CompiledQuery,
-    chunkSize?: number,
```

Comment:
optionality here was wrong; all streaming query builders always defaulted to `100`.